### PR TITLE
Add Weights and Biases Logging

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
This PR aims to add basic [**Weights and Biases**](https://wandb.ai/site) Metric Logging by appending to the existing `MetricLogger` Class defined in [**`utils.py`**](https://github.com/salesforce/BLIP/blob/25511dc4bb06ed98d21b11eeaa1aa0659a6b4102/utils.py#L92) with minimal changes while supporting Multiple GPU logging with torch distributed. 

The changes can be summarized as follows :-

1. Pass the `args` to the `MetricLogger` which if the `--use_wandb` is set to `True`, creates and run and logs metrics using the `update` function.
2. Add 3 extra arguments namely `--use_wandb`, `--wandb_project` and `--wandb_entity` which can be used to specify whether to use wandb, the name of the project to be used (`"BLIP"` by default) and name of the entity to be used.

---

Here are some graphs and runs that were logged using this logger.

| Image        | Description    |
|--------------|-----------|
| BLIP-loss | A W&B Panel showcasing the loss as logged by `metric_logger.update(loss=loss.item())`  |
| Screenshot 2022-02-07 at 12 26 06 PM | The corresponding wandb config containing all parameters provided using `argparse`, logged using `self.wandb_run = wandb.init(..., config=args)` |

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
Request for Review @LiJunnan1992. I've already signed the CLA but it's not reflecting.

**Comment by LiJunnan1992** [comment]
@SauravMaheshkar Thanks for contributing! However, because this feature is not critical, we do not plan to merge it so that we can keep the training code simpler.
